subsys/debug/Kconfig

```diff
@@ -67,12 +67,35 @@ config ASAN_NOP_DLCLOSE
 
 config UBSAN
 	bool "Build with undefined behavior sanitizer"
-	depends on ARCH_POSIX
```

Comment:
The commit message states:
> then you need the ubsan library routines which are only
> present for posix architecture or when using picolibc.

So should this instead be:
```
	depends on ARCH_POSIX || PICOLIBC
```
instead of being completely removed ?